src/tls/certificate.c

```diff
@@ -68,6 +70,23 @@ certificate_get_credentials (Certificate *self)
   return self->creds;
 }
 
+time_t
+certificate_get_expiry (Certificate *self)
+{
+  gnutls_x509_crt_t *crt_list;
+  unsigned int crt_list_size;
+
+  int ret = gnutls_certificate_get_x509_crt (self->creds, 0, &crt_list, &crt_list_size);
+  assert (ret == GNUTLS_E_SUCCESS);
```

Comment:
Could this rather use `err()`? AFAIK it's entirely plausible/possible to "compile out" assert(). If this fails, then the code below would access uninitialized memory.

Reply:
this is a case of "that should *really* never fail", but OK.

Reply:
I think you were referring to the line below this one, and indeed, that one could be different.  I've changed it to an `if`/`errx()` as suggested.

Reply:
I actually did mean exactly this line. If it's a Should Not Fail™, then ok :grinning: